### PR TITLE
 #1369 Change connector definition's prop from list to array

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [X] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1369 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by wu87988622** [comment]
/cc @chia7712 QA build failed can you support  or what do I need to do?
```
Task :ohara-kafka:compileJava FAILED
/ohara/ohara-kafka/src/main/java/com/island/ohara/kafka/connector/json/SettingDefinition.java:201: error: cannot find symbol
          .valueType(Type.LIST)
                         ^
  symbol:   variable LIST
  location: class Type
1 error

FAILURE: Build failed with an exception.
```

**Comment by wu87988622** [comment]
> /cc @chia7712 QA build failed can you support or what do I need to do?
> 
> ```
> Task :ohara-kafka:compileJava FAILED
> /ohara/ohara-kafka/src/main/java/com/island/ohara/kafka/connector/json/SettingDefinition.java:201: error: cannot find symbol
>           .valueType(Type.LIST)
>                          ^
>   symbol:   variable LIST
>   location: class Type
> 1 error
> 
> FAILURE: Build failed with an exception.
> ```

Thanks for @saivirtue help~~~

**Comment by chia7712** [comment]
> QA build failed can you support or what do I need to do?

The patch I gave to you is out-of-date :(

File: ohara-manager/client/src/components/pages/PipelinePage/Connectors/connectorUtils.js [hunk 1]
```diff
@@ -280,20 +280,18 @@ export const renderer = props => {
   const dataType = ['String'];
   const tableActions = ['Up', 'Down', 'Delete'];
 
-  return formData.map(data => {
+  const domWithValueType = params => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/PipelinePage/Connectors/connectorUtils.js]: I'd rename this to `renderWithValueType`

File: ohara-manager/client/src/components/pages/PipelinePage/Connectors/connectorUtils.js [hunk 2]
```diff
@@ -333,7 +331,30 @@ export const renderer = props => {
...
+              handleColumnRowDown={handleColumnRowDown}
+            />
+          </FormGroup>
+        );
+      default:
+        return null;
+    }
+  };
+
+  const domWithReference = params => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/PipelinePage/Connectors/connectorUtils.js]: renderWithReference

**Comment by wu87988622** [comment]
retry

**Comment by wu87988622** [comment]
retry

**Comment by wu87988622** [comment]
retry

**Comment by wu87988622** [comment]
retry

**Comment by wu87988622** [comment]
retry

**Pull request closed without merging**